articles/build-an-outlook-clone-using-react-hooks/index.md

```diff
@@ -0,0 +1,881 @@
+
+Once we have the server running, we can check via Postman or just go to http://localhost:3000/emailbody. To get a specific email body, we can send the *id* as a parameter. The URL is modified as follows: http://localhost:3000/emailbody?id=1. This returns the first email body.
+
+### Let's implement the hooks we just learned
+
+In the earlier article, we have discussed the logic to go about implementing the project from scratch. In this article, we will give more importance to understanding the hooks and performing optimization.
+
+### Explaination: *App.js*
+
+We declare functional components and hooks need to be declared within the function. We use various state variables. They are
+1. emails
+2. dualPanel
+3. id
+4. body
+5. searchField
+6. readList
+7. fav
+8. forRead
+9. forSearch
+10. forFav
+11. unread
+
+These are the same state variables as earlier. This time they are declared as and when used. This is a powerful feature. Declare state variables as and when required.
+
+Initially, on componentDidMount, the first useEffect is called, where we fetch the list. Initially, the id is set to an empty string. When there is a change in id, we will make an API call to the localhost to fetch the body. The body state is set. getId is the callback function that is passed as a prop to Email component and we set the id.
+
+Earlier, we had to send a callback function to modify the readlist. In this case, we can directly send the useState list. We send readlist and setReadList as props to *Email.js*. In *Email.js*, we setReadList on an onClick event.
+
+We have also used createContext. We declare a global const EmailContext. This is wrapped around the child component. We send the emails as the value to be received at the child component. In the child component, we use the hook useContext to receive the prop email.
+
+### Styled-components
+
+This is by far the easiest library to use for styling. There are various alternatives like inline styling and tachyons. But styled-components enables us to use our CSS knowledge. Let's say we want to style a button element. The inline styling would be
+
+```jsx
+<button style={{borderRadius:'1%'}}>Hey there</button>  
+```
+Using styled-components, on the other hand, would allow us to maintain a tidy code. Let's look at styling the button using styled-components.
+
+All we have to do is create a constant and define our CSS styling within the constant. We can also create classes and ids similar to CSS.
+
+```jsx
+const ButtonStyle = styled.button {
+    border-radius: 1%;
+    color: white;
+    font-family: "Helvetica Neue", Helvetica;
+    .app: {
+        font-size: 12px;
+    };
+}
+
+function App() {
+    <ButtonStyle className='app'>
+        <button>Hey there</button>  
+    </ButtonStyle>
+}
+
+```
+Inline styling decreases code readability. Therefore, styledComponents is good.
```

Comment:
> Inline styling decreases code readability. Therefore, styledComponents is good.

Remove "Therefore, styledComponents is good." and move " Inline styling decreases code readability." in front of this sentence: "Using styled-components, on the other hand, would allow us to maintain a tidy code. Let's look at styling the button using styled-components."